app/Database/InProgCourseQueries.hs

```diff
@@ -0,0 +1,70 @@
+--| UberGoal: go from T.Text -> [Entity Lecture] / [Entity Tutorial] without multiple DB queries.
+
+import Data.Tuple.Select as Tup -- selecting first, second and third elements of tuples (fall, spring, year)
+
+-- | Step 1: single query for lecture or tutorial. Produces list of all 
+-- | lectures/tutorials respectively, whcih are not separated by session
+
+-- | Queries db for all matching records with lecture or tutorial code of this course
+lectureQuery :: T.Text <- [Entity Lecture]
```

Comment:
Also, you may want to look at how we're converting `Entity Lecture -> Lecture` (there's an `entityVal` helper function defined somewhere).